runtime/oti/j9accessbarrier.h

```diff
@@ -171,13 +171,13 @@ typedef struct J9IndexableObject* mm_j9array_t;
 	(void)0)
 #define J9OBJECT__PRE_OBJECT_STORE_ADDRESS(vmThread, object, address, value) \
 	((void)0, \
-	(OMR_GC_ALLOCATION_TYPE_SEGREGATED == (J9VMTHREAD_JAVAVM((vmThread)))->gcAllocationType) ? \
+	((OMR_GC_WRITE_BARRIER_TYPE_SATB || OMR_GC_WRITE_BARRIER_TYPE_SATB_AND_OLDCHECK) == (J9VMTHREAD_JAVAVM((vmThread)))->gcWriteBarrierType) ? \
```

Comment:
this is actually not correct. the 'or' condition should be between to equality comparisons

Reply:
fixed